packages/grid/x-data-grid/src/tests/selection.DataGrid.test.tsx

```diff
@@ -83,6 +83,48 @@ describe('<DataGrid /> - Selection', () => {
     });
   });
 
+  describe('props: checkboxSelection = false (single selection), with keyboard events', () => {
+    it('should select one row at a time on Shift + Space', () => {
+      render(<TestDataGridSelection />);
+      getCell(0, 0).focus();
+      fireEvent.keyDown(document.activeElement || document.body, { key: ' ', shiftKey: true });
```

Comment:
Why `document.activeElement || document.body`? This will snooze the ESLint rule "Don't use document.activeElement as a target for keyboard events.". You can pass the cell element as bellow:

https://github.com/mui-org/material-ui-x/blob/1d82bea672afc1ece26739a47b2de7ae298de9bc/packages/grid/x-data-grid-pro/src/tests/clipboard.DataGridPro.test.tsx#L91-L93

Reply:
Not really, it is to follow the documentation :)
 > This will also test that the element in question can even receive keyboard events

https://testing-library.com/docs/guide-events#keydown

Reply:
The element is passed to improve readability: https://github.com/mui-org/material-ui/pull/20945

> This will also test that the element in question can even receive keyboard events
 
This check is embedded into the custom `fireEvent` we use. See https://github.com/mui-org/material-ui/blob/69b7e4d911e8ff99d17a74330496ae69e7cc0ade/test/utils/createRenderer.tsx#L622

Reply:
I've done the modifications, but I'm not really convinced that it improves readability

The core team has element which are all meaning full, leading to testing code such as 
```js
fireEvent.keyDown(button, ...);
fireEvent.keyDown(firstTab, ...);
```

We are navigating between cells so most of the components with focus are cells which only differ by there coordinates which is translated by
```js
fireEvent.keyDown(cell01, ...);
fireEvent.keyDown(cell23, ...);
```

Reply:
Take as example the following test case:

```tsx
expect(getActiveCell()).to.equal('8-1');
fireEvent.keyDown(document.activeElement!, { key: 'ArrowDown' });
fireEvent.keyDown(document.activeElement!, { key: 'ArrowRight' });
fireEvent.keyDown(document.activeElement!, { key: 'ArrowLeft' });
fireEvent.keyDown(document.activeElement!, { key: 'ArrowUp' });
```

Can you easily spot where the focus is after the last `keyDown`? 😛